plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptionsVariationSelector.php

```diff
@@ -17,16 +17,313 @@ class AddToCartWithOptionsVariationSelector extends AbstractBlock {
+			$option_label = $is_taxonomy ? $item->name : $item;
+
+			if ( ! $is_taxonomy || in_array( $option_value, $options, true ) ) {
+				$selected_attr = $is_taxonomy
+					? selected( sanitize_title( $selected ), $option_value, false )
+					: selected( $selected, $option_value, false );
+
+				/**
+				 * Filter the variation option name.
+				 *
+				 * @since 9.7.0
+				 *
+				 * @param string     $option_label    The option label.
+				 * @param WP_Term|string|null $item   Term object for taxonomies, option string for custom attributes.
+				 * @param string     $attribute_name  Name of the attribute.
+				 * @param WC_Product $product         Product object.
+				 */
+				$filtered_label = apply_filters(
+					'woocommerce_variation_option_name',
+					$option_label,
+					$is_taxonomy ? $item : null,
+					$attribute_name,
+					$product
+				);
+
+				$html .= sprintf(
+					'<option value="%s" %s>%s</option>',
+					esc_attr( $option_value ),
+					$selected_attr,
+					esc_html( $filtered_label )
+				);
+			}
+		}
+
+		return $html;
+	}
+
+	/**
+	 * Render variation form.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $attributes Block attributes.
+	 * @return string Rendered form HTML
+	 */
+	private function render_variation_form( $product, $attributes ): string {
+		$variation_attributes = $product->get_variation_attributes();
+		if ( empty( $variation_attributes ) ) {
+			return '';
+		}
+
+		wp_enqueue_script( 'wc-add-to-cart-variation' );
+
+		$variations = $this->get_variations_data( $product );
+		if ( empty( $variations ) ) {
+			return '';
+		}
+
+		return $this->get_form_html( $product, $variations, $variation_attributes );
+	}
+
+	/**
+	 * Get variations data.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @return array|false
+	 */
+	private function get_variations_data( $product ) {
+		/**
+		 * Filter the number of variations threshold.
+		 *
+		 * @since 9.7.0
+		 *
+		 * @param int        $threshold Maximum number of variations to load upfront.
+		 * @param WC_Product $product   Product object.
+		 */
+		$get_variations = count( $product->get_children() ) <= apply_filters( 'woocommerce_ajax_variation_threshold', 30, $product );
+		return $get_variations ? $product->get_available_variations() : false;
+	}
+
+	/**
+	 * Get form HTML.
+	 *
+	 * @param WC_Product $product Product instance.
+	 * @param array      $variations Variations data.
+	 * @param array      $variation_attributes Variation attributes.
+	 * @return string Form HTML
+	 */
+	private function get_form_html( $product, $variations, $variation_attributes ): string {
+		$variations_json = wp_json_encode( $variations );
+		$variations_attr = function_exists( 'wc_esc_json' ) ? wc_esc_json( $variations_json ) : _wp_specialchars( $variations_json, ENT_QUOTES, 'UTF-8', true );
+
+		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => 'wc-block-add-to-cart-form' ) );
```

Comment:
Why do we need the wrapper `<div>`? Won't this block be inside the Add to Cart with Options block anyway? :thinking:

Reply:
Good call! I removed it in the commit [b3097c1](https://github.com/woocommerce/woocommerce/pull/53947/commits/b3097c15757e5c4227d09ce34c60751b2b002ed9)